### DISCUSSION TITLE
Get the version in GitHub Actions to use in other build steps

### DISCUSSION THREAD

**Discussion by Peaj** [adamralph/minver#939] [Q&A] [answered]
Is there any way to extract the generated version number when building with GitHub Actions?
I would like to use the version number in the artifact name.

**Comment by adamralph** [comment, marked as answer]
Yes, see https://github.com/adamralph/minver#can-i-use-minver-to-version-software-which-is-not-built-using-a-net-sdk-style-project

**Comment by bording** [comment]
You can see an example of that being done here: https://github.com/libgit2/libgit2sharp.nativebinaries/blob/master/.github/workflows/ci.yml#L82-L86

And then the value is used here: https://github.com/libgit2/libgit2sharp.nativebinaries/blob/master/.github/workflows/ci.yml#L92

**Reply by Peaj** [reply]
Thanks this solution works.
I thought there might be some way to extract the version instead of running MinVer again (e.g. an Environment variable)
This way it would not be necessary to config it in two places and it could spare a bit of build time.

**Reply by adamralph** [reply]
This FAQ I posted above described a solution which avoids configuring MinVer in two places. You pass all your configuration to the CLI tool as arguments, and then you set the MINVERVERSIONOVERRIDE env var to the version which the CLI tool outputs, and then the value of that env var is used as-is by the MinVer packaged installed in your .NET projects.

**Reply by Peaj** [reply]
I see. I guess I overread that. That would reduce the need to configure it twice in theory. But If I want to support local builds I still have to configure MinVer in the project itself. This ability to support local builds is one of the reasons a chose to use MinVer over other solutions.
Installing MinVer on the runner pushed my build time from 3:15 to 5:30 btw.
I think it might be possible to set environment variables from MSBuild and then access them from GitHub actions.
I will see if I can get this to work.

**Reply by adamralph** [reply]
> But If I want to support local builds I still have to configure MinVer in the project itself. This ability to support local builds is one of the reasons a chose to use MinVer over other solutions.

This is the reason I perform local builds with a build script, and all my CI server does is call the build script. Then I have the same outputs both locally and in CI.

> Installing MinVer on the runner pushed my build time from 3:15 to 5:30 btw.

There's nothing I can do to help that I'm afraid. That's just the .NET CLI.

> I think it might be possible to set environment variables from MSBuild and then access them from GitHub actions.

I guess that would work too. But I would always favour running a build script both locally and in CI, to be able to replicate full builds locally, rather than performing special actions in CI that don't occur locally (aside from things like pushing artifacts, etc.)

**Reply by Peaj** [reply]
Yes you are right that it is best to have the whole build script work locally.
Pushing artifacts is exactly the step I am trying to change.
I want the artifact name to contain the version number so I can differentiate them when downloading.
Thanks for you help. I will see if I can figure out a way to make the environment vars work.
Until then having MinVer as an additional build step is a valid solution.

**Reply by adamralph** [reply]
Why can't the build script produce the artifact with the correct name?

**Reply by Peaj** [reply]
It kind of does produce the correct folder output.
But the artifact-upload github action does not get the name from the path.
It has to be set manually. But it might still work manually read and set it from the path somehow.

**Reply by adamralph** [reply]
I guess I can't really comment much further without knowing the details of your use case, but FWIW, I also use `actions/upload-artifact`, and while the filename of the artifact zip has a [static, unversioned name](https://github.com/adamralph/simple-exec/blob/af9086245ad7c394a2836447c7116d637c3294a6/.github/workflows/ci.yml#L37), the contents themselves are unambiguously versioned, e.g NuGet package file names, and that comes from the use of MinVer within the build script.

**Reply by Peaj** [reply]
Yes thats the same in my case. I just dont like the artifact having a static name because its also the name of the zip you download and you end up having many zips called "Package (3)" etc. so I use the version number in the artifact name to distinguish the archives.
